server/jetstream_cluster.go

```diff
@@ -3094,8 +3094,10 @@ func (js *jetStream) processClusterDeleteStream(sa *streamAssignment, isMember,
 		sa.Group.node.Delete()
 	}
 
-	// Cleanup in case account does not exist or node was nil
-	if sacc := s.SystemAccount(); sacc != nil {
+	// This is a stop gap cleanup in case
+	// 1) the account does not exist (and mset couldn't be stopped) and/or
+	// 2) node was nil (and couldn't be deleted)
+	if sacc := s.SystemAccount(); sacc != nil && (!stopped || sa.Group.node == nil) {
```

Comment:
Would reverse so in happy path do not need server lock to lookup system account. Also deref on sa.Group.node should be under js lock, so maybe capture above.

Reply:
fixed (also modified earlier sa.Group.node access)